js/views/AddressBarIndicators.js

```diff
@@ -0,0 +1,105 @@
+import _ from 'underscore';
+import loadTemplate from '../utils/loadTemplate';
+import baseVw from './baseVw';
+import { isMultihash } from '../utils';
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    super(options);
+
+    this._state = {
+      hide: true,
+      url: '',
+      ...options.initialState || {},
+    };
+  }
+
+  updateVisibilityBasedOn(addressBarText) {
+    if (typeof addressBarText !== 'string') {
+      throw new Error('Please provide a valid address bar as string.');
+    }
+
+    const viewOnWebState = {
+      hide: true,
+      url: '',
+    };
+
+    const urlParts = this.getUrlParts(addressBarText);
+
+    if (urlParts.length > 1 && isMultihash(urlParts[0])) {
+      const supportedPages = ['store', 'home', 'followers', 'following'];
+      const currentPage = urlParts[1];
+
+      if (supportedPages.includes(currentPage)) {
+        const obDotCom = 'http://openbazaar.com';
+        const peerID = urlParts[0];
+
+        if (currentPage === 'store') {
+          // app: '/peerID/store/' => web: '/store/peerID/'
+          viewOnWebState.url = `${obDotCom}/store/${peerID}`;
+
+          if (urlParts.length === 3) {
+            // app: '/peerID/store/slug' => web: '/store/peerID/slug'
+            const slug = urlParts[2];
+            viewOnWebState.url = `${viewOnWebState.url}/${slug}`;
+          }
+        } else {
+          // app: '/peerID/(home|followers|following)' =>
+          // web: '/store/(home|followers|following)/peerID'
+          viewOnWebState.url = `${obDotCom}/store/${currentPage}/${peerID}`;
+        }
+      }
+    }
+
+    viewOnWebState.hide = viewOnWebState.url.length === 0;
+    this.setState(viewOnWebState);
+  }
+
+  getUrlParts(url) {
+    if (typeof url !== 'string') {
+      throw new Error('Please provide a valid url as a string.');
+    }
+
+    const urlParts = url.startsWith('ob://') ?
+      url.slice(5)
+      .split(' ')[0] :
+      url.split(' ')[0];
+
+    return urlParts.split('/');
+  }
+
+  className() {
+    return 'addressBarIndicators';
+  }
+
+  getState() {
+    return this._state;
+  }
+
+  setState(state, replace = false) {
```

Comment:
Can you use the state handlers built into baseVw here?

Reply:
Yup, we extend our views from the baseVw and methods like `setState` are already implemented there, so no need to redeclare them.

The one exception is if you need to do some pre/post processing, but then you would only do you custom code and pipe it through the base class method.

(FWIW, in some views you will see setState/getState implemented. That's because we moved them to the baseVw after we had just been repeating them for a while and we haven't gone back to clean them up, since they might have subtle nuances and it would need some decent testing. We took the approach that going forward, we'll rely on the baseVw and already implemented ones, we'll clean up when we have bandwidth/inspiration)

Reply:
> (FWIW, in some view...

@rmisio: That's precisely why I ended up putting that in there - because I saw you guys doing it elsewhere and wanted to adhere to how you're doing things.

Thanks for clearing that up so I can now remove it and not have the hairy eyebrow.